src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureKeyVaultReference/AzureKeyVaultSecretProvider.cs

```diff
@@ -154,5 +167,42 @@ private void UpdateNextRefreshableSecretFromCache()
                 _nextRefreshTime = null;
             }
         }
+
+        private DateTimeOffset? GetCacheExpirationTimeForSecret(string key, CachedKeyVaultSecret cachedSecret, bool success)
+        {
+            DateTimeOffset? refreshSecretAt = null;
+
+            if (success)
+            {
+                cachedSecret.RefreshAttempts = 0;
```

Comment:
It's not a good idea to make changes inside a `Get` method. Change it outside.

Reply:
Alternatively, you change the method name to something like `ExtendCacheExpirationTimeForSecret`.

Reply:
Thanks, updated.